### DISCUSSION TITLE
Incremental Static Regeneration

### DISCUSSION THREAD

**Discussion by alfajrip** [vercel/next.js#53153] [Help] [answered]
### Summary

Hello Community,

I am using Incremental Static Regeneration approach with Next.Js v13.2.4.

I have promotional data with the default disabled, when it is not active the promo page will not be displayed, or 404. and when the promo is activated it will display the page along with the promo data. But, it doesn't work as expected, when the promo is activated what should be displayed is the promo page, but now what I'm experiencing is always showing 404. In this case, the ISR doesn't seem to regenerate the page I activated. But, when I update data other than boolean type, ISR runs fine.

I hope y'all here can help to solve this issue, thank you in advance.

### Additional information

```js
export async function getStaticPaths() {
	const { data: promotions } = await PromotionServices.getPromotions('v1');

	const paths = promotions.map((promotion) => ({
		params: {
			slug: promotion.slug,
		},
	}));

	return {
		paths,
		fallback: 'blocking',
	};
}

export async function getStaticProps(context) {
	const { slug } = context.params;

	const { data: promotions } = await PromotionServices.getPromotions('v1');

	const isPublished = promotions.some((promotion) => promotion.slug === slug && promotion.active);

	if (!isPublished) {
		return {
			notFound: true,
		};
	}

	const { data: promotion } = await PromotionServices.getPromotionBySlug('v1', slug);

	return {
		props: {
			promotion,
		},
		revalidate: 10,
	};
}
```

**Comment by icyJoseph** [comment, marked as answer]
Add a revalidate to the `notFound` branch:

```ts
if (!isPublished) {
		return {
			notFound: true,
                         revalidate: 10
		};
	}
```

**Reply by alfajrip** [reply]
thank you for your answer @icyJoseph , works as expected.
